### ISSUE TITLE
Progressive Language Feature

### ISSUE THREAD

**Issue by ms-mialingvo** [asterics/AsTeRICS-Grid#344]
Option to assign a certain level to a button. The button is faded out until that level is set in the settings. With higher levels the number of buttons displayed gradually increases, allowing the user to learn a large grid gradually with no changes to the position of the symbols.

**Comment by klues** [comment]
implemented first version:
* select element level here: ![Image](https://github.com/user-attachments/assets/b8045935-43aa-46b0-a9b1-04a39a794c36)
* select shown level here: ![Image](https://github.com/user-attachments/assets/dfd75f8a-1475-4d65-8af5-902de0de5c0c)
* little markers showing level in edit mode: ![Image](https://github.com/user-attachments/assets/3a1a8d31-2053-4a3a-afc1-81e0fecc58de)

Level is also possible to copy via property brush.
Test at https://gridbeta.asterics-foundation.org/

**Comment by ms-mialingvo** [comment]
YAY!!! 🥰 🥳 Happy!

Tested on Safari Mac only so far, it's working perfectly there and looking good!

It should be also in the `general` tab in the main settings. 
The term is `progressive language`. So in the edit grid element it should be `progressive language level` and in the main settings `default progressive language level`.

**Comment by klues** [comment]
Great! 🙂

> It should be also in the general tab in the main settings.

Hmm, I think `progressive language` makes most sense in tab `language`, doesn't it? Probably your thought is that it should be `General`, because it's also in `General` tab in edit element?! However, I still think that `language` fits better and if something should be changed it should be a `language` tab in edit element (maybe some time in the future, since now it would only contain this single element).

> So in the edit grid element it should be progressive language level and in the main settings default progressive language level.

`Progressive language level` is very long and will somehow break the layout of the edit element tab. If you think it's important, I can try to change the layout (make the input elements smaller), but I think just `language level` should be enough?!

In settings I also think that `default progressive language level` doesn't make much sense, because it's not the default that is somehow applied to other elements, but just the level determining which elements are shown. So maybe something like `shown language level`? And maybe the heading `Progressive language`?

**Comment by klues** [comment]
updated:
* element hints for level are bold, if this element will be shown with the current setting
* color of element hints automatically have good contrast to element background color

**Comment by ms-mialingvo** [comment]
> Probably your thought is that it should be General, because it's also in General tab in edit element?! However, I still think that language fits better and if something should be changed it should be a language tab in edit element (maybe some time in the future, since now it would only contain this single element).

Yeah, exactly because of that. I can't imagine anything else that could fit into a `language` tab on single element level, and as you pointed out, it's not worth having a tab for just one setting.

> Progressive language level is very long and will somehow break the layout of the edit element tab. If you think it's important, I can try to change the layout (make the input elements smaller), but I think just language level should be enough?!

Hm, I guess in that case `vocabulary level` would be clearer for people not knowing that progressive language is.

> So maybe something like shown language level? And maybe the heading Progressive language?

Yes, `shown vocabulary level` should work. And that heading.

Just a reminder, as it's not written here but we only discussed that in the meeting: It would be neat if there was an action for an element that lets people switch between full vocabulary and whatever-is-set-in-the-settings-vocabulary-level.

**Comment by ms-mialingvo** [comment]
Oh, and I think it's fine if the option goes under `Advanced options/settings`, it won't be used that often I think.

**Comment by klues** [comment]
ok:
* renamed "language level" to "vocabulary level"
* moved it to "advanced" in edit element
* improved translations in settings, e.g. giving options like `up to level 5` for `shown vocabulary level`

Please test at gridbeta.
@arasaac-dga please translate new strings to spanish.

**Comment by ms-mialingvo** [comment]
Thanks :)
